luskan/src/main/java/org/opencadc/luskan/ws/StorageTapService.java

```diff
@@ -0,0 +1,224 @@
+import ca.nrc.cadc.tap.schema.InitDatabaseTS;
+import ca.nrc.cadc.uws.server.impl.InitDatabaseUWS;
+import ca.nrc.cadc.vosi.AvailabilityPlugin;
+import ca.nrc.cadc.vosi.AvailabilityStatus;
+import ca.nrc.cadc.vosi.avail.CheckDataSource;
+import ca.nrc.cadc.vosi.avail.CheckException;
+import ca.nrc.cadc.vosi.avail.CheckResource;
+
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.Principal;
+import javax.security.auth.Subject;
+import javax.security.auth.x500.X500Principal;
+import javax.sql.DataSource;
+import org.apache.log4j.Logger;
+
+/**
+ *
+ * @author adriand
+ */
+public class StorageTapService implements AvailabilityPlugin {
+
+    private static final Logger log = Logger.getLogger(StorageTapService.class);
+    
+    private static final Principal TRUSTED = new X500Principal("cn=servops_4a2,ou=cadc,o=hia,c=ca");
+
+    private static final String TAP_TEST = "select schema_name from tap_schema.schemas11 where schema_name='tap_schema'";
+    private static final String UWS_TEST = "select jobID from uws.Job limit 1";
+
+    private String appName;
+
+    public StorageTapService() {
+    }
+
+    @Override
+    public void setAppName(String appName) {
+        this.appName = appName;
+    }
+    
+    @Override
+    public AvailabilityStatus getStatus() {
+        boolean isGood = true;
+        String note = "service is accepting queries";
+        try {
+            String state = getState();
+            if (RestAction.STATE_OFFLINE.equals(state)) {
+                return new AvailabilityStatus(false, null, null, null, RestAction.STATE_OFFLINE_MSG);
+            }
+            if (RestAction.STATE_READ_ONLY.equals(state)) {
+                return new AvailabilityStatus(false, null, null, null, RestAction.STATE_READ_ONLY_MSG);
+            }
+
+            String tapDsName = DataSourceProviderImpl.getDataSourceName(null, "tapadm");
+            DataSource tapadm = DBUtil.findJNDIDataSource(tapDsName);
+            InitDatabaseTS tsi = new InitDatabaseTS(tapadm, null, "tap_schema");
+            tsi.doInit();
+
+            String uwsDsName = DataSourceProviderImpl.getDataSourceName(null, "uws");
+            DataSource uws = DBUtil.findJNDIDataSource(uwsDsName);
+            InitDatabaseUWS uwsi = new InitDatabaseUWS(uws, null, "uws");
+            uwsi.doInit();
+
+            // create the TAP schema
+            InitLuskanSchemaContent lsc = new InitLuskanSchemaContent(tapadm, null, "tap_schema");
+            lsc.doInit();
+
+            // run a couple of queries
+            CheckResource cr = new CheckDataSource(tapDsName, TAP_TEST);
+            cr.check();
+
+            cr = new CheckDataSource(uwsDsName, UWS_TEST);
+            cr.check();
+
+            LocalAuthority localAuthority = new LocalAuthority();
+            RegistryClient reg = new RegistryClient();
+
+            // TODO - These do not work for intTest unless the testing environment deploys these services.
```

Comment:
It should be possible to make these work now that cadc-tomcat has a cert injection scheme. But if not just leave them as-is and I'll figure it out next week.

Reply:
They don't work because they are checked against local services which are not deployed.